gobblin-modules/gobblin-kafka-common/src/main/java/org/apache/gobblin/source/extractor/extract/kafka/KafkaSource.java

```diff
@@ -221,6 +221,16 @@ public String apply(KafkaTopic topic) {
             }
           }), state);
 
+      for (KafkaTopic topic : topics) {
+        if (topic.getTopicSpecificState().isPresent()) {
+          if (topicSpecificStateMap.containsKey(topic.getName())) {
+            topicSpecificStateMap.get(topic.getName()).addAllIfNotExist(topic.getTopicSpecificState().get());
+          } else {
+            topicSpecificStateMap.put(topic.getName(), topic.getTopicSpecificState().get());
```

Comment:
Is there any chance this state gets mutated later? Is there a need for a deep copy?
If a deep copy is required then you can replace the if/else block with 
`topicSpecificStateMap.computeIfAbsent(topic.getName(), k -> new State()).addAllIfNotExist(.getTopicSpecificState().get());`

Reply:
Good point, the way I'm using it I don't think it would be mutated, but it's probably better to do that to be safe.